### PR TITLE
[Map] Update camera position after map stops moving

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
I've observed that the map sometimes hangs while panning/zooming the map. 

From looking at the code it appears that,
 *  `dragInteractions` are only used to toggle the location lock
 *  `cameraMoves` continuously invokes the new position but only the last one gets consumed by the app.

(Please correct me if the assumptions I've made are incorrect)

So I've created this experimental PR which changes a few behaviours: 
 * Trigger `dragInteractions` only once when the map starts to move.
 * Trigger `cameraMoves` after the map pan/zoom is finished.

Pros:
 * Much smoother map interactions since number of invocations are significantly reduced.
 * Updated `targetSdkVersion` to 30 since the above problem was behind the map hanging issue.

@gino-m @scolsen Please let me know what you think :)

### CODE REVIEW

**Comment by gino-m** [comment]
@shobhitagarwal1612 This is great! Afaict your assumptions are correct. In that case, perhaps we can rename the two streams to `startDragEvents` or similar, and `cameraMovedEvents` (with accessors renamed accordingly)?

Please let me know once the PR is finalized and I'll take a closer look. Thanks for this!

File: gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java [hunk 1]
```diff
@@ -470,25 +470,18 @@ private int parseColor(@Nullable String colorHexCode) {
...
-    Point target = fromLatLng(gmsCameraPosition.target);
-    CameraPosition position = new CameraPosition(target, gmsCameraPosition.zoom);
-    cameraMoves.onNext(position);
-    if (cameraTargetBeforeDrag != null
-        && !gmsCameraPosition.target.equals(cameraTargetBeforeDrag)) {
-      dragInteractions.onNext(target);
+  private void onCameraMoveStarted(int reason) {
+    cameraChangeReason = reason;
+    if (reason == REASON_GESTURE) {
+      dragInteractions.onNext(true);
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java:484]: If the value isn't meaningful consider using `NIL` in our custom `Nil` class instead.

**Comment by shobhitagarwal1612** [comment]
The build is failing remotely because the container is running out of memory. We might have to upgrade the machine type of the container.

**Comment by gino-m** [comment]
> The build is failing remotely because the container is running out of memory. We might have to upgrade the machine type of the container.

Is that a matter of updating the config, or does something else need to be done?

**Comment by shobhitagarwal1612** [comment]
While debugging, I found that gradle is terminating due to lack of memory. 
https://github.com/google/ground-android/pull/904#issuecomment-864364666

I'm not sure why this started happening randomly. But my guess is that we need to build another docker image with an upgraded machine type to fix this problem.

https://stackoverflow.com/questions/58536639/google-cloud-build-custom-machine-type

**Review by gino-m — DISMISSED** [review]
Nice work, this is much clearer, thank you!

**Comment by gino-m** [comment]
@googlebot rescan

**Comment by gino-m** [comment]
@shobhitagarwal1612 Are [these](https://00f74ba44b068330ff460cbcd5c8ae6a62c31cc833-apidata.googleusercontent.com/download/storage/v1/b/artifact_bucket/o/map-perf-dcf64eae-40c5-41ff-aaa2-59170f50672c%2Freports%2Ftests%2FtestStagingUnitTest%2Findex.html?jk=AFshE3U3PnLykBM5xq5nR_pqKyX_TgfJ4ShBSfUTcHl_KtYP4AkE4vmr4JTRr0_uTizjkPg5GYaOkfEQpYaW-w5EMldJlQ9eCoq5t4lmejFjTOBK75GWHgjlDQsixK3WkYdmil46eywGgs_Em1elNuQPqbi2Jx7V-WssSLYxDubF6aPIMF7rDmE-g0PQdfI_tpJzVIkvb_rcK-DJBtFF9hwg-Xt0FqcvloViFmbVRkWuSZA1aobh-wu5jKGkbilst33jw0LtoL2M69VQTIDyGPjubkEpmx7VoYeW8B85YSp9jguSvrm2A9zYTI6mPc-hDEKYsSXmu7wB_OrXU4NmaPJd2yzpKPR9UhXkhOyhrHDKUhKRnIWXXIG1YhRd1Cb76ioh2dWL8EqULNLNpyonXej8K6KAodsR56CDW5PJera-rVI4i6MQKFh70k7_Ss8YyqUQQZKdQEpT3z1usNxUhByGwA1OAFRPc6DhEsKVVwgBQ6fN-4nT9U3vTH_to1Ms0ysYt642lfMMt_w3N74zlfp4GVXGoyy-95YEtVFsmG96p9jEOx816QEKzQC2yonCRaf9BjNGp06AZTFSrDAfV89zRIBSazmSdR-C5us3fwOOiOlN1cBCPTy9Ato4eTvzePl93d04iiDLUQeNl-mKiTLzOLjOUXaqIOsbWN5QrZrjGCbTSY8fvs6cQwIFnQKQsbDF42wLkytwOqQXz9ZuzsXt_xWiVCAhNdMOMlLNgLW2flVjFPZD3MJzo0spY8NFi0TfvB_qxOwht_4S_sJ5dym8DuSw5Er8uqdFgs3B0S_-PwSU-T5U93VUIfuvNzzskrDzO-_WKjGQJoyB3qLZYlVb6aRB_CuL6hQ_8j6NtHwIKG_gESItEpXeuxDsZdGOMuqTBB7Z2biN9SZIvq2NBySSFPWlllO4d3TCtUjW8fWfTRbLVtevzyqgaqVwmBymCwmw4mZQYNCytGPTaQ_C3RgNZR3_D9omLK9yryXPm3LoWsI-QRGmEaaHW1cp3ajfoCEna2bh2aNBXh8_yhnLM1QXNEtP4HZdFeLdUBPLrRuC2kFQ_P2tbK-LKg&isca=1) actual failing tests? I saw them when rerunning checks after disabling instrumented tests:

**Comment by shobhitagarwal1612** [comment]
The tests were failing because of API level 30. Reverted that commit it is still unresolved.

**Review by gino-m — APPROVED** [review]
LGTM!

**Pull request merged**